catalog/spatial/stanag4559/spatial-stanag4559-source/src/test/java/org/codice/ddf/spatial/ogc/stanag4559/catalog/source/data/DAGGenerator.java

```diff
@@ -0,0 +1,440 @@
+            node.id = id;
+            id++;
+        }
+    }
+
+    /**
+     * Set the UCO.Edges of the DAG according to the STANAG 4459 Spec.  This requires the ids
+     * of the Nodes to be set in DFS order.
+     *
+     * @param root  - the root node of the graph (NSIL_PRODUCT)
+     * @param graph - the graph representation of the DAG
+     */
+    private static void setUCOEdges(Node root, Graph<Node, Edge> graph) {
+        Stack<Node> stack = new Stack<>();
+        Stack<Node> visitorStack = new Stack<>();
+        stack.push(root);
+
+        while (!stack.isEmpty()) {
+            Node currNode = stack.pop();
+            if (!visitorStack.contains(currNode)) {
+                visitorStack.push(currNode);
+                for (Edge edge : graph.edgesOf(currNode)) {
+
+                    Node source = graph.getEdgeSource(edge);
+                    Node target = graph.getEdgeTarget(edge);
+
+                    // Remove if statement?
+                    if (edge != null && source != null && target != null) {
+                        edge.start_node = source.id;
+                        edge.end_node = target.id;
+                        stack.push(target);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Constructs the NSIL_PRODUCT subgraph of the NSIL_ALL_VIEW.  This method sets builds the NSIL_PRODUCT
+     * with all optional nodes ( NSIL_APPROVAL, etc.) as well as all MANDATORY attributes for these NODES
+     * according to the STANAG 4459 spec.
+     *
+     * @param orb            - a reference to the orb to create UCO objects
+     * @param graph          - the graph representation of the DAG
+     * @param numRelatedFile - the number of NSIL_RELATED_FILE's to create.  This number is unbounded
+     *                       according to the specification.
+     * @return a Node[] that contains a reference to the root, NSIL_SECURITY, and NSIL_CARD that are used
+     * in other subgraphs.
+     */
+    private static Node[] constructNSIL_PRODUCT(ORB orb, Graph<Node, Edge> graph,
+            int numRelatedFile) {
+        List<String> product_nodes = Arrays.asList("NSIL_APPROVAL",
+                "NSIL_FILE",
+                "NSIL_STREAM",
+                "NSIL_METADATA_SECURITY",
+                "NSIL_CARD",
+                "NSIL_SECURITY");
+        List<Node> nodeProductNodes = getEntityListFromStringList(product_nodes, orb);
+
+        Node[] nodeArray = new Node[3];
+
+        Node root = constructRootNode(orb);
+        nodeArray[0] = root;
+        graph.addVertex(root);
+        Node attribute;
+
+        for (Node node : nodeProductNodes) {
+            graph.addVertex(node);
+            graph.addEdge(root, node);
+
+            if (node.attribute_name.equals("NSIL_SECURITY")) {
+                nodeArray[1] = node;
+            } else if (node.attribute_name.equals("NSIL_CARD")) {
+                nodeArray[2] = node;
+            }
```

Comment:
I usually try to make my static Strings static final in case they need to be called from other places in the code.